Improve GPU resource cache management

emxsys opened the pull request:
### Description of the Change
Adds an interface to GpuResourceCache and its backing MemoryCache to advance or retard the normal aging of a cache entry.  Typically users of the cache would an _aging multiplier_ on volatile cache entries so they age faster and evict sooner.

### Why Should This Be In Core?
Provides desirable and expected behaviors when viewing the globe such that recently viewed map tiles remain in the cache in preference to temporal text entries like view coordinates.

### Benefits
- Provides an orderly eviction of volatile resources.
- The resource eviction policies are under the control the resource creators. 
- The existing GpuResourceCache and MemoryCache function interfaces are unchanged.

### Potential Drawbacks
- A 100x aging multiplier has been applied to text textures.
- A 10x aging multiplier has been applied to surface shape tiles.

### Applicable Issues
Closes #497

emxsys commented:
Debugging code is available in the MemoryCache for the testing. Set DEBUG=true to dump the MemoryCache to the console during evictions.

pdavidc commented:
@emxsys and I just discussed this: There's a minor naming change to make, but otherwise this looks good to me.
@markpet49 Do you have any questions or concerns?

emxsys commented:
Updated pull request to use 'agingFactor' term and removed debug code.  Tested with Shapefiles example to assert correct behavior with Text, ShapeSurfaceTiles, and TiledImageLayer.

pdavidc commented:
Looks good to me now.
